raysect/optical/observer/sampler1d.pyx

```diff
@@ -112,25 +163,28 @@ cdef class MonoAdaptiveSampler1D(FrameSampler1D):
         if pixels != frame.length:
             raise ValueError('The pixel geometry passed to the frame sampler is inconsistent with the pipeline frame size.')
 
-        min_samples = max(self.min_samples, frame.samples.max() / self.ratio)
+        min_samples = max(self._min_samples, <int>(frame.samples.max() / self._ratio))
         error = frame.errors()
+        mean = frame.mean_mv  # does memoryview initialisation check before the loop
```

Comment:
Sorry, didn't spot this earlier.

https://cython.readthedocs.io/en/latest/src/userguide/source_files_and_compilation.html#compiler-directives

This change wasn't necessary. The @cython.initializedcheck(False) was missing, if you add this, all the memoryview checks are avoided. Could you please revert this change and add the above decorator. Same goes for the other places.

Reply:
It's done.